### DISCUSSION TITLE
This branch is 1 commit ahead of xxx:main

### DISCUSSION THREAD

**Discussion by digger-yu** [community/community#53389] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello everyone, I am a github newbie
When I commit the PR on github, I get this message on my project 
This branch is 1 commit ahead of xxxxxx:main.
When I click the sync fork button on the page
my remote branch will become
This branch is 2 commits ahead of xxxxxx:main.
I think it is correct to strictly record each commit, but feel that the ease of use here is not very good
I've searched google for ways to keep them both in line, but it feels complicated

In this case
How can I make my remote branch consistent with the project's master branch in the easiest way possible?

**Comment by airtower-luna** [comment, marked as answer]
> How can I make my remote branch consistent with the project's master branch in the easiest way possible?

Warning to start with: Consistent means *identical*, any commits/changes you made would be lost. :warning: 

If that's what you want, you can do the following, starting from a local clone of your repository:

1. `git fetch` the upstream repository (add it as a remote first if needed)
2. `git checkout` the right upstream branch
3. Push with `--force` to the branch in your fork. That's the step where you'll overwrite  changes you might have made, so double check first. :warning: 

See also https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork, which describes the process in more detail, and also lists an alternative using `gh`. Mind that merging with diverged histories will result in a merge commit (as you got via the web UI), which will be another different commit, if you want your branch to be identical to upstream you need to use the branch from upstream directly.

**Reply by digger-yu** [reply]
thank you very much

**Reply by airtower-luna** [reply]
Glad I could help! Please don't forget to mark the answer. :slightly_smiling_face:
